Review of specs 2025-06-02

padajo opened the pull request:
There are a few changes suggested, and a few comments made in the commit as well.

Comments may be missing some context or understanding, but hopefully this should help with clarity.

Read commit message before looking at suggested changes.

padajo committed:
Suggested changes to registry/1.0.md

padajo committed:
Suggested changes to registry-versioning/1.0md

Including adding in the definition for "xsd:" which was missing.

padajo committed:
registry-process-resources/1.0.md

Not clear what is meant by "the URL" in the first section. It could be read as suggesting that the "process" defined cannot be changed because the URL cannot change.

Maybe link to versioning document here to clarify?

padajo committed:
machine-readable-data-licenses/1.0.md

The wording suggests that once a license has been published into the Registry, it is "immutable". This makes sense but it isn't clear how a license is removed from the Registry, should that ever be needed.

Needs clarity on wording to explain what happens when a license is removed, revoked or amended.

padajo committed:
Suggested changes to data-catalog-records/1.0.md and role-based-access-control/1.0.md

Definition of dataset is unclear, so improvement suggested.

Not clear in documentation why Open Energy is used as a reference to explain decisions made. Needs clarification.

padajo committed:
Suggested changes to member-identity-digital-certificates/1.0.md

Need to add in how long server certificates last because it's not clear in the documentation.

Have removed where it says that certificates do not have to be checked for revocation, because that's implying a lack of risk when it does exist.

Have also removed the specification of ACME for automatic renewal because even in the case of ACME, it's a manual process to install a certificate and we haven't yet agreed that ACME is the correct process.

padajo committed:
No changes for baseline-tls-configuration/1.0.md

padajo committed:
Suggested changes to oauth-with-member-identity-certificates/1.0.md

Have added "(PAR)" to one heading to ensure that where PAR is used elsewhere, it is referencing this.

Could do with a bit more explanation about how OAuth is being used, and maybe an image for how the flow works because it's not entirely clear here (flow section).

croesus commented on docs/registry-versioning/1.0.md:
```diff
@@ -89,7 +90,7 @@ Each change may only include changes from one scheme (or be schemeless, making c
-: URLs of documents which show evidence of the governance process. Where possible, these document should be stored in the Registry.
+: URLs of documents which show evidence of the governance process. Where possible, these documents should be stored in the Registry.
 
 `ib1:registrySnapshot`
```
Is there a specification for the format/generation process for this file?

bensummers replied:
My recollection is that we discussed this evidence, but didn't have enough info to know what they would look like, or where they would be hosted. So the result is this vague statement.

croesus replied:
OK, something to revisit in future, probably driven by the need to actually version a Registry for real

padajo committed:
No changes for message-delivery-to-applications/1.0.md

It isn't clear until the end that revocation of linked permissions needs to be cascaded. This part should be made much clearer in withdrawal-of-permission/1.0.md

padajo committed:
Provenance needs clearer examples I think. I wonder if it should be broken out into its own spec?
